src/Ocelot/Request/Middleware/DownstreamRequest.cs

```diff
@@ -42,37 +44,35 @@ public DownstreamRequest(HttpRequestMessage request)
 
     public HttpRequestMessage ToHttpRequestMessage()
     {
-        var uriBuilder = new UriBuilder
-        {
-            Port = Port,
-            Host = Host,
-            Path = AbsolutePath,
-            Query = RemoveLeadingQuestionMark(Query),
-            Scheme = Scheme,
-        };
-
-        _request.RequestUri = uriBuilder.Uri;
+        var uri = CreateUri();
+        
+        _request.RequestUri = uri;
         _request.Method = new HttpMethod(Method);
         return _request;
     }
 
     public string ToUri()
+    {
+        return CreateUri().AbsoluteUri;
+    }
+
+    public override string ToString()
+    {
+        return ToUri();
+    }
+
+    private Uri CreateUri()
     {
         var uriBuilder = new UriBuilder
         {
             Port = Port,
             Host = Host,
-            Path = AbsolutePath,
+            Path = string.IsNullOrEmpty(ServicePathPrefix) ? Path : $"{ServicePathPrefix.TrimEnd('/')}/{Path.TrimStart('/')}",
```

Comment:
Trimming slashes is not advisable for Ocelot's current functionality for routing features and bugfixes which are related to slash in the path: last slash, omitting slash, Catch All paths.
We recommend verifying this by creating acceptance with and enabling a real service discovery provider.

Reply:
The `TrimStart` merely fixes the case where the `Path` already starts with a slash - without the ` TrimStart`, we could have two slashes (a hardcoded slash is part of the interpolated string).